lxd/container.go

```diff
@@ -748,22 +748,24 @@ func containerCreateInternal(d *Daemon, args containerArgs) (container, error) {
 
 func containerConfigureInternal(c container) error {
 	// Find the root device
-	for _, m := range c.ExpandedDevices() {
-		if m["type"] != "disk" || m["path"] != "/" || m["size"] == "" {
-			continue
-		}
+	localDevices := c.LocalDevices()
+	_, rootDiskDevice := containerGetRootDiskDevice(localDevices)
+	if rootDiskDevice["pool"] == "" {
+		expandedDevices := c.ExpandedDevices()
+		_, rootDiskDevice = containerGetRootDiskDevice(expandedDevices)
+	}
```

Comment:
I thought last time we established that if you had a device:

root:
  type: disk
  path: /
  size: 100GB

In the containers config and a device called:

defintely-another-name:
  type: disk
  path: /

in the profiles then you end up with both devices:
- root
- definitely-another-name
in the expanded devices...